plugins/module_utils/network/sonic/config/route_maps/route_maps.py

```diff
@@ -0,0 +1,2073 @@
+        'route_maps',
+    ]
+
+    route_maps_uri = 'data/openconfig-routing-policy:routing-policy/policy-definitions'
+    route_map_uri = route_maps_uri + '/policy-definition={0}'
+    route_map_stmt_uri = route_map_uri + '/statements/statement={1}'
+    route_map_stmt_base_uri = route_map_uri + '/statements/statement={1}/'
+    route_maps_data_path = 'openconfig-routing-policy:policy-definitions'
+
+    set_community_rest_names = {
+        'additive': 'openconfig-routing-policy-ext:ADDITIVE',
+        'local_as': 'openconfig-bgp-types:NO_EXPORT_SUBCONFED',
+        'no_advertise': 'openconfig-bgp-types:NO_ADVERTISE',
+        'no_export': 'openconfig-bgp-types:NO_EXPORT',
+        'no_peer': 'openconfig-bgp-types:NOPEER',
+        'none': 'openconfig-bgp-types:NONE'
+    }
+
+    set_extcomm_rest_names = {
+        'rt': 'route-target:',
+        'soo': 'route-origin:'
+    }
+
+    def __init__(self, module):
+        super(Route_maps, self).__init__(module)
+
+    def get_route_maps_facts(self):
+        """ Get the 'facts' (the current configuration)
+
+        :rtype: A dictionary
+        :returns: The current configuration as a dictionary
+        """
+        facts, _warnings = Facts(self._module).get_facts(self.gather_subset,
+                                                         self.gather_network_resources)
+        route_maps_facts = facts['ansible_network_resources'].get('route_maps')
+        if not route_maps_facts:
+            return []
+        return route_maps_facts
+
+    def execute_module(self):
+        """ Execute the module
+
+        :rtype: A dictionary
+        :returns: The result from module execution
+        """
+        result = {'changed': False}
+        warnings = list()
+
+        existing_route_maps_facts = self.get_route_maps_facts()
+        commands, requests = self.set_config(existing_route_maps_facts)
+        if commands and len(requests) > 0:
+            if not self._module.check_mode:
+                try:
+                    edit_config(self._module, to_request(self._module, requests))
+                except ConnectionError as exc:
+                    self._module.fail_json(msg=str(exc), code=exc.errno)
+            result['changed'] = True
+        result['commands'] = commands
+
+        changed_route_maps_facts = self.get_route_maps_facts()
+
+        result['before'] = existing_route_maps_facts
+        if result['changed']:
+            result['after'] = changed_route_maps_facts
+
+        result['warnings'] = warnings
+        return result
+
+    def set_config(self, existing_route_maps_facts):
+        """ Collect the configuration from the args passed to the module,
+            collect the current configuration (as a dict from facts)
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        want = remove_empties_from_list(self._module.params['config'])
```

Comment:
For states other than 'deleted', required arguments `action` and `sequence_num` could be validated with internal ansible methods (Ref: https://github.com/ansible-collections/dellemc.enterprise_sonic/blob/main/plugins/module_utils/network/sonic/config/l3_acls/l3_acls.py#L583)

Reply:
I have added this validation.

Reply:
Thanks for adding a common method to perform custom value validations.